src/core/server.js

```diff
@@ -676,6 +676,17 @@ Server.prototype._handleResponse = function(message) {
     const result = message.result;
     const responseEvent = 'response_' + command;
 
+    if (command === 'submit' && result.engine_result === 'tefPAST_SEQ' &&
+        request.reconnectionsCount > 0
+    ) {
+      // handle case when request reached the server, but response was lost
+      // and request was automatically repeated
+      result.engine_result = 'tesSUCCESS';
```

Comment:
What do you think of having a default timeout and making request.js return a `tejTimeout` error?